readthedocs/oauth/services/gitlab.py

```diff
@@ -0,0 +1,222 @@
+
+        :param fields: dictionary response of data from API
+        :rtype: RemoteOrganization
+        """
+        try:
+            organization = RemoteOrganization.objects.get(
+                slug=fields.get('path'),
+                users=self.user,
+                account=self.account,
+            )
+        except RemoteOrganization.DoesNotExist:
+            organization = RemoteOrganization.objects.create(
+                slug=fields.get('path'),
+                account=self.account,
+            )
+            organization.users.add(self.user)
+
+        organization.name = fields.get('name')
+        organization.account = self.account
+        organization.url = '{url}/{path}'.format(
+            url=self.adapter.provider_base_url, path=fields.get('path')
+        )
+        if fields.get('avatar'):
+            organization.avatar_url = '{url}/{avatar}'.format(
+                url=self.adapter.provider_base_url,
+                avatar=fields['avatar']['url'],
+            )
+        organization.json = json.dumps(fields)
+        organization.save()
+        return organization
+
+    def setup_webhook(self, project):
+        """Set up GitLab project webhook for project
+
+        :param project: project to set up webhook for
+        :type project: Project
+        :returns: boolean based on webhook set up success
+        :rtype: bool
+        """
+        session = self.get_session()
+
+        # See: http://doc.gitlab.com/ce/api/projects.html#add-project-hook
+        data = json.dumps({
+            'id': 'readthedocs',
+            'push_events': True,
+            'issues_events': False,
+            'merge_requests_events': False,
+            'note_events': False,
+            'tag_push_events': True,
+            'url': 'https://{0}/gitlab'.format(settings.PRODUCTION_DOMAIN),
+        })
+        resp = None
+        try:
+            repositories = RemoteRepository.objects.filter(
+                clone_url=project.vcs_repo().repo_url
+            )
+            assert repositories
+            repo_id = repositories[0].get_serialized()['id']
+            resp = session.post(
+                '{url}/api/v3/projects/{repo_id}/hooks'.format(
+                    url=self.adapter.provider_base_url,
+                    repo_id=repo_id,
+                ),
+                data=data,
+                headers={'content-type': 'application/json'}
+            )
+            if resp.status_code == 201:
+                log.info('GitLab webhook creation successful for project: %s',  # noqa
+                         project)
+                return True
+        except (AssertionError,  RemoteRepository.DoesNotExist) as ex:
+            log.error('GitLab remote repository not found', exc_info=ex)
+        except RequestException as ex:
+            pass
+        else:
+            ex = False
+
+        log.error('GitLab webhook creation failed for project: %s',  # noqa
+                  project, exc_info=ex)
```

Comment:
This try block could shortened, it seems like there is extraneous handling of `ex`